File: fixtures/gold/stripe-app/regex-routes.js
// Regex-literal masking regression (Loop 294). Both lines below used to be
// provider-level FALSE NEGATIVES before the scanner's regex masker:
//  - `/^\/\//` masked the rest of its own line as a `//` comment, hiding
//    the require() that follows it on the same line;
//  - `/foo\/*bar/` opened a phantom `/*` block comment that blacked out
//    every line below it in the file.
const isProto = /^\/\//.test(target); const stripe = require('stripe');
const splat = /foo\/*bar/;
const client = new stripe.Stripe(process.env.STRIPE_KEY);

File: fixtures/gold/stripe-app/regex_routes.rb
# Positive control (Loop 303): a regex literal must not black out real
# code after it — on the same line or below.
module StripeRegexRoutes
  def self.frag?(raw)
    raw.match?(/v1#frag/); key = ENV['STRIPE_SECRET_KEY']
    key
  end
end

require 'stripe'
